kitsune/users/views.py

```diff
@@ -435,6 +422,36 @@ def verify_token(self, token, **kwargs):
         # deserializing https://bugs.python.org/issue17909
         return json.loads(jws.payload.decode('utf-8'))
 
+    def process_events(self, payload):
+        fxa_uid = payload.get('sub')
+        profile_q = Profile.objects.filter(fxa_uid=fxa_uid)
```

Comment:
You can do here:
```python
try:
    profile = Profile.objects.get(fxa_uid=fxa_uid)
except Profile.DoesNotExist:
    profile = None
```

Then you can check if the profile var has a value. If not you can skip the computation and just save the event in the db without any profile

Reply:
This can also remove some nesting below.

Reply:
You can also bundle this to a handy utility like here (probably not in the jinja helpers):
https://github.com/mozilla/mozillians/blob/master/mozillians/common/templatetags/helpers.py#L267